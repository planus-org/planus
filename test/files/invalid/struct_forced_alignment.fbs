struct Align1 (force_align: 1) {
  value: uint16;
}

struct Align2 (force_align: 2) {
  value: uint32;
}

struct Align3 (force_align: 3) {
  value: uint8;
}

struct Align4 (force_align: 4) {
  value: uint64;
}

struct Inner (force_align: 8) {
  value: uint8;
}

struct Align8 (force_align: 4) {
  value: Inner;
}
