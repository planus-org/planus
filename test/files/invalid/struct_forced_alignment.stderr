error: Alignment of struct cannot be forced to lower
  ┌─ test/files/invalid/struct_forced_alignment.fbs:1:16
  │
1 │ struct Align1 (force_align: 1) {
  │                ^^^^^^^^^^^^^^ This attribute tries to force the alignment of the struct
2 │   value: uint16;
  │          ------ However the minimum alignment of this type is 2

error: Alignment of struct cannot be forced to lower
  ┌─ test/files/invalid/struct_forced_alignment.fbs:5:16
  │
5 │ struct Align2 (force_align: 2) {
  │                ^^^^^^^^^^^^^^ This attribute tries to force the alignment of the struct
6 │   value: uint32;
  │          ------ However the minimum alignment of this type is 4

error: Alignment must be a power of two
  ┌─ test/files/invalid/struct_forced_alignment.fbs:9:16
  │
9 │ struct Align3 (force_align: 3) {
  │                ^^^^^^^^^^^^^^

error: Alignment of struct cannot be forced to lower
   ┌─ test/files/invalid/struct_forced_alignment.fbs:13:16
   │
13 │ struct Align4 (force_align: 4) {
   │                ^^^^^^^^^^^^^^ This attribute tries to force the alignment of the struct
14 │   value: uint64;
   │          ------ However the minimum alignment of this type is 8

error: Alignment of struct cannot be forced to lower
   ┌─ test/files/invalid/struct_forced_alignment.fbs:21:16
   │
21 │ struct Align8 (force_align: 4) {
   │                ^^^^^^^^^^^^^^ This attribute tries to force the alignment of the struct
22 │   value: Inner;
   │          ----- However the minimum alignment of this type is 8

